src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,428 @@
+                **SubscriptionId:** 'a9f0536e-15cc-4d1f-79a4-08d79acf8bba'
+                **Source Repository :**  'https://github.com/maestro-auth-test/maestro-test1'
+                **Target Repository :**  'https://github.com/maestro-auth-test/maestro-test2'
+                **Branch Name :**  '38'
+                **Error Message :**  'build: 14	Unexpected error processing action: Validation Failed'
+                **Method :** 'UpdateAssetsAsync'
+                **Action :** 'Updating assets for subscription: ee8cdcfb-ee51-4bf3-55d3-08d79538f94d'
+                **Last seen :** 'Maestro.Data.RepositoryBranchUpdateHistoryEntry'
+
+                The Regex : (\w+): '([^,\n]+)'(?:, |\)) captures (subscriptionId: 'a9f0536e-15cc-4d1f-79a4-08d79acf8bba', method: 'UpdateAssetsAsync', errorMessage: 'build: 14	Unexpected error processing action: Validation Failed')
+                the above string and puts it into two groups. 
+
+                group[1]            group[2]
+                subscriptionId      a9f0536e-15cc-4d1f-79a4-08d79acf8bba
+                method              UpdateAssetsAsync
+                errorMessage        build: 14	Unexpected error processing action: Validation Failed
+
+                This is put into a dictionary, where group[1] is put in as key and group[2] as value.
+        */
+        private static Dictionary<string, string> ParseIssueCommentBody (string body)
+        {
+            var arguments = new Dictionary<string, string>();
+            var matches = CommentMarker.Matches(body);
+
+            foreach (Match match in matches) {
+                var group = match.Groups;
+                arguments.Add(group[1].Value, group[2].Value);
+            }
+            return arguments;
+        }
+
+        private string ParseUpdateAssetsAsync(RepositoryBranchUpdateHistoryEntry updateHistoryError)
+        {
+            var description = "";
+            string subscriptionId = GetSubscriptionId(updateHistoryError.Arguments);
+            Guid subscriptionGuid = GetSubscriptionGuid(subscriptionId); 
+            var subscription = (from sub in
+                _context.Subscriptions
+                where sub.Id == subscriptionGuid
+                select sub).FirstOrDefault();
+            // Subscription might be removed, so if the subscription does not exists then the error is no longer valid. So we can skip this error.
+            if (subscription == null)
+            {
+                Logger.LogInformation("SubscriptionId:" + subscriptionId + " has been deleted for the repository : ");
+                return string.Empty;
+            }
+            description =
+                $"[marker]: <> (subscriptionId: '{subscriptionId}', method: '{updateHistoryError.Method}', errorMessage: '{updateHistoryError.ErrorMessage}')"
```

Comment:
You should be able to use a verbatim string `@$""` to avoid all the concatenation

Reply:
Here and in other places with heavy use of concatenation

Reply:
I redid it. I misunderstood this concatenation part, I must have clarified this earlier. Thank you for pointing it out

Reply:
fixed this finally!